Design/trivially_exportable_citations.md

```diff
@@ -0,0 +1,183 @@
+Documentation tooling should be produced to automatically populate citations on documentation pages.
+
+The instructions of what to cite thet are shown when you launch Mantid will be removed.
+They will be replaced with instructions of how to export the citations from a workspace.
+
+## Design
+
+This is largely based on what Savu does to solve the same problem.
+
+### Citation data type
+
+A new data type (`Citation`) will be created to store citations.
+This type will be exported to Python.
+
+As conversion between citation formats is non trivial, the algorithm developer is required to supply the citation in all formats.
+
+The bare minimum for this type will be something similar to the following:
+```cpp
+class Citation
+{
+  Citation();
+
+  std::string description;
+  std::string url;
+  std::string doi;
+  std::string bibtex;
+  std::string endnote;
+
+  void loadNexus(...);
+  void saveNexus(...) const;
+};
+```
+
+The fields here mirror those of [`NXcite`](http://download.nexusformat.org/doc/html/classes/base_classes/NXcite.html#nxcite).
+The following restrictions apply to ensure that in all cases somehting citable is allowed to exist:
+  - `description` is always optional (this isn't needed for citation, but gives insight as to why this citation is relevant)
+  - if `bibtex` is provided `endnote` must also be provided, and vice-versa (BibTex and Endnote contain essentially the same information, they can both be created if one can be. BibTex and Endnote do not imply a DOI is minted)
+  - if `doi` is provided, `url`, `bibtex` and `endnote` must all be provided (BibTex and Endnote can be generated from DOIs)
+  - if none of `doi`, `bibtex` or `endnote` are provided, `url` must be provided (there must be something there, even if this isn't citable a URL is better than nothing)
+
+Some helper "toString" methods may be beneficial.
+
+### Constructing a citation
+
+Citations themselves will store the same data as `NXcite` however some helper functionality will be implemented to make creating them easier.
+
+`Citation` could have several constructors that take `struct`s that define information required for each citation type.
+Such `struct`s may look something like (by no means complete, just to give an idea):
+```cpp
+
+struct OnlineCitation
+{
+  std::string url;
+  std::string whenAccessed;
+};
+
+struct BookCitation
+{
+  std::vector<std::string> authors;
+  std::string title;
+  std::string publisher;
+  std::string pages;
+};
+
+struct PaperCitation
+{
+  std::vector<std::string> authors;
+  std::string title;
+  std::string journal;
+  int year;
+  std::string doi;
+};
+
+```
+
+In the constructor for the relevant `struct`, `Citation` would then call `to_bibtex` and `to_endmode` functions to obtain citations in the required formats.
+DOI and URL fields can be directly extracted where possible.
+
+As an example, creating a citation could be done like so:
+```cpp
+auto const savu = Citation(PaperCitation{
```

Comment:
Thanks for the addition of an example. I think removing the requirement for the developer to format the citation properly is a good move.